libraries/lib-module-manager/PluginDescriptor.h

```diff
@@ -0,0 +1,157 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  PluginDescriptor.h
+
+  Split from PluginManager.h
+
+**********************************************************************/
+
+#pragma once
+#include <wx/string.h>
+
+#include "PluginInterface.h"
+#include "wxArrayStringEx.h"
+
+///////////////////////////////////////////////////////////////////////////////
+//
+// PluginDescriptor
+//
+///////////////////////////////////////////////////////////////////////////////
+
+typedef enum : unsigned {
+   PluginTypeNone = 0,          // 2.1.0 placeholder entries...not used by 2.1.1 or greater
+   PluginTypeStub =1,               // Used for plugins that have not yet been registered
+   PluginTypeEffect =1<<1,
+   PluginTypeAudacityCommand=1<<2,
+   PluginTypeExporter=1<<3,
+   PluginTypeImporter=1<<4,
+   PluginTypeModule=1<<5,
+} PluginType;
+
+class PluginDescriptorXMLTagHandler;
+
+// TODO:  Convert this to multiple derived classes
+//! Represents either a PluginProvider or a loaded plug-in and caches some
+//! information about it
+class MODULE_MANAGER_API PluginDescriptor
+{
```

Comment:
While you are touching this class, please do these things too:

- Use in-class member initializers so that the body of the constructor shrinks to nothing.
- Make sure no other members of non-class types are uninitialized.
- Remove the friend declaration, because long-distance friendships are bad for understanding code.  Make Load() and Set() public so you can.